# 전체 데이터 로딩

In [87]:
import numpy as np

# 디렉토리 변경으로 '..'을 통해 로드했음. 변경된 디렉토리 구조 참고하시길.
total_data = np.load('../data/data_/data_all4040.npy')
print(total_data.shape[:])

(8575, 40, 40, 4)


# 이미지 전처리
-> 이미지 1시간 간격으로 끊어서 저장만 하겠음, 사전 저장된 모듈 활용

In [88]:
import data_preprcss as dpp
# 리로딩 안하면 굉장히 골치 아픔
from importlib import reload
reload(dpp)

<module 'data_preprcss' from 'c:\\주 폴더\\학교\\2024\\세특\\학술제\\real 학술제\\제출 파일\\ClimateAI\\AcademicFestival\\Main\\data_preprocessing_codes\\data_preprcss.py'>

In [89]:
# 일단 작은 규모의 테스트 데이터 먼저 넣어보자..
temp_list1 = total_data[:30*12]
print('테스트 데이터 규모:', temp_list1.shape[:])

input_imgs1, output_imgs1 = dpp.splitImgs_with_1Hour(temp_list1)
print(f'''입력 데이터 규모: {input_imgs1.shape[:]}
출력 데이터 규모: {output_imgs1.shape[:]}
''')

테스트 데이터 규모: (360, 40, 40, 4)
입력 데이터 규모: (30, 11, 40, 40, 4)
출력 데이터 규모: (30, 1, 40, 40, 4)



In [90]:
# 조금 더 크면서도 애매한 걸로
temp_list2 = total_data[:360]
print('테스트 데이터 규모:', temp_list2.shape[:])

input_imgs2, output_imgs2 = dpp.splitImgs_with_1Hour(temp_list2)
print(f'''입력 데이터 규모: {input_imgs2.shape[:]}
출력 데이터 규모: {output_imgs2.shape[:]}
''')

테스트 데이터 규모: (360, 40, 40, 4)
입력 데이터 규모: (30, 11, 40, 40, 4)
출력 데이터 규모: (30, 1, 40, 40, 4)



이후 여러 가지 배치 검증 테스트 결과, 알고리즘 정상 작동함

# 본격적인 데이터 전처리

이론상 예상 데이터 개수 : 8575 = 360 x 23 + 295 에서 295개 데이터는 버려짐 -> 실제와 일치  
  
  
이론상 데이터셋(1p) 개수 : 몰라요  

실제 데이터셋(1p) 개수 : 690 -> 와우

In [91]:

splited_inputs, splited_outputs = dpp.splitImgs_with_1Hour(total_data)

In [92]:
np.save("../data/data_/data_1hours_inputs_MEGA", splited_inputs)
np.save("../data/data_/data_1hours_outputs_MEGA", splited_outputs)

In [93]:
print('입력 shape:', splited_inputs.shape[:])
print('출력 shape', splited_outputs.shape[:])

입력 shape: (690, 11, 40, 40, 4)
출력 shape (690, 1, 40, 40, 4)
